src/App.js

```diff
@@ -36,10 +38,30 @@ class App extends Component {
     return Math.floor((new Date() - date) / (365 * 60 * 24 * 1000 * 60));
   }
 
+  handleSort = (e) => {
+    if(this.state.sortBy === e.target.innerText.toLowerCase()){
+      this.setState({
+        orderDesc: !this.state.orderDesc
+      });
+    }else{
+      this.setState({
+        sortBy: e.target.innerText.toLowerCase(),
+        orderDesc: true
+      });
+    }
+    this.forceUpdate();
+  }
+
   render() {
     let rows = [];
     let options = [];
 
+    if(this.state.sortBy === 'tech'){
+      this.state.technologies.sort((a, b) => this.state.orderDesc ? a.name > b.name : a.name < b.name )
+    }else{
+      this.state.technologies.sort((a, b) => this.state.orderDesc ? a.released > b.released : a.released < b.released )
```

Comment:
Don't mutate state, perhaps you can create a copy of `technologies` and then update state via `setState`.
And this logic should be in click handler itself.